File: problems/1109.航班预订统计.py
#
# @lc app=leetcode.cn id=1109 lang=python
#
# [1109] 航班预订统计
#

# @lc code=start



class Solution(object):
    def corpFlightBookings(self, bookings, n):
        """
        :type bookings: List[List[int]]
        :type n: int
        :rtype: List[int]
        """
        nums = [0] * n
        for l ,r , v in bookings:
            nums[l-1] += v
            if r < n:
                nums[r] -=v
        
        for i in range(1,n):
            nums[i] += nums[i-1]
        
        return nums
        # 还原数据
# @lc code=end

File: problems/1109.航班预订统计.java
/*
 * @lc app=leetcode.cn id=1109 lang=java
 *
 * [1109] 航班预订统计
 */

// @lc code=start
class Solution {
    /**
     * 区间修改：对区间内的数加N
     * @param bookings
     * @param n
     * @return
     */
    public int[] corpFlightBookings(int[][] bookings, int n) {
        int[] diff = new int[n+1];
        // [i,j] diff[i]+val, diff[j+1] -val
        
        for (int[] order : bookings) {
            int i = order[0] -1;
            int j = order[1] -1;
            int val = order[2];
            diff[i] += val;
            diff[j+1] -=val;
        }
        
        // // 还原数组
        for (int i = 1; i < n; i++) {
            diff[i] += diff[i-1];
        }

        return Arrays.copyOf(diff, n);
    }
}
// @lc code=end

